userspace/libsinsp/sinsp.h

```diff
@@ -75,28 +75,21 @@ using namespace std;
 #include "utils.h"
 
 #ifndef VISIBILITY_PRIVATE
+// Some code defines VISIBILITY_PRIVATE to nothing to get private access to sinsp
 #define VISIBILITY_PRIVATE private:
+#define VISIBILITY_PROTECTED protected:
```

Comment:
This is confusing.  If `VISIBILITY_PRIVATE` isn't defined, you set both `VISIBILITY_PRIVATE` and `VISIBILITY_PROTECTED`, otherwise you set only `VISIBILITY_PROTECTED`?

Seems like you want:

    #if !defined(VISIBILITY_PRIVATE)
    #    define VISIBILITY_PRIVATE private:
    #endif 

    #if !defined(VISIBILITY_PROTECTED)
    #    define VISIBILITY_PROTECTED protected:
    #endif

Reply:
You're right, but then I need to go change all the clients of this to have `#define VISIBILITY_PROTECTED` and that just makes this weirdness worse.  So I did it this way and hopefully we can tear this out faster.